index.js

```diff
@@ -98,15 +110,25 @@ class ServerlessIotLocal {
 
   _createMQTTBroker() {
     const { host, port, httpPort } = this.options
-    this.mqttBroker = createMQTTBroker({
+    const { redisHost, redisPort, redisDB } = this.options
+
+    const ascoltatore = {
+      redisHost,
```

Comment:
i misled you earlier, sorry, i forgot ascolatore is just a part of the mosca opts object

can you change it back so the opts are pass-through? This way broker.js doesn't need to know about option names like redisHost, redisPort

something like:

```js
// index.js
this.mqttBroker = createMQTTBroker({
    host,
    port,
    http: {
      host,
      port: httpPort,
      bundle: true
    },
    // ascolatore
    backend: {
       host: redisHost,
       port: redisPort,
       ...
    }
})
```

```js
// broker.js
function createBroker(opts) {
  // deep merge 
  const moscaOpts = _.merge({}, moscaSettings, opts)
  ...
}

```

Reply:
I can certainly move things around but remember that the original implementation has the `persistence` declaration as part of broker.  It seems strange that broker would know that persistence is Redis but would not know about the redis options.  In fact, the original `moscaSettings` and `ascoltatore` both lived in broker and were also responsible for settings like `return_buffers`.  `ascoltatore` and `mosca` seems like good broker options without exposing mosca to index and that make sense to me.  The broker should know how to construct mosca options.  index should not know how to construct mosca options.

Thoughts?

Reply:
right, the original implementation made a whole lot of assumptions. What do you think about avoiding the problem altogether, and having an optional `mosca` options object under the `serverless-iot-local` key in serverless.yml:

```yaml
custom:
  serverless-iot-local:
    mosca:
      http: ...
      backend:
        db: 12
        host: ...
```

then pass it through directly to broker, _.merge with the defaults and pass through to `mosca`. Advanced users can then customize to their heart's content, and everyone else can just stick with the current defaults. Otherwise in a month or two someone's going to want to customize some other option to mosca or ascolatore or redis or whatnot and we'll be back here adding code to support it.

Reply:
This seems like a lot of flexibility to solve a simple problem of redis options not being exposed.  The backend could be anything, mongo, redis or any of the other backend types that mosca supports.  You're suggesting that the entire mosca configuration be exposed. 
 How do you plan on validating the mosca options?  This is getting out of control for my immediate needs.  A forked repo is starting to look appealing.

Reply:
sure, i understand, it's a lot of unneeded flexibility for your use case. It's probably simplest for you to use your fork, as you said. If you decide to do that and still want to submit a PR for the describeEndpoint fix, happy to accept it (or I'll fix it myself if you don't want to waste time on it, let me know).

my issue with the extra options: today we add support for `redisHost`, `redisPort`, and tomorrow someone wants to support some other deep mosca option. Then we'd probably need to revert this and go with the pass-through option or keep adding options indefinitely (with custom name mappings like yours)

Reply:
@mvayngrib A little frustration leaked out into my last comment; sorry for that.  Why don't you accept the proposed PR and let the next developer that needs more flexibility submit a PR for it.  Alternatively further flexibility could be added later.  I just don't have the bandwidth to keep going right not.

As a compromise, I could a single redis config option-

`custom:
  serverless-iot-local:
    redis:
      host:
      port:
      db:`

This way the redis config could be changed but the options are isolated in a single redis config object.

Reply:
sounds good, let's compromise with the `redis` config option
btw, i'm using AWS Device SDK without certs. See the changes in my [fork](https://github.com/aws/aws-iot-device-sdk-js/commit/71d26f6a8c3fddb4e8a84e78874a43dc04fb56c8)

Reply:
lol, ha!  Thanks for sharing!  More changes incoming...